### PR TITLE
KOGITO-4350 Add sibling DMN sample to BPMN2 in Archetypes

### PR DESCRIPTION
**Pull request by tarilabs**
Ref: https://issues.redhat.com/browse/KOGITO-4350

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by tarilabs** [comment]
```
[INFO] ------------------------------------------------------------------------

[INFO] Reactor Summary for Kogito Runtimes 2.0.0-SNAPSHOT:

[INFO] 

[INFO] Kogito Runtimes .................................... SUCCESS [  0.257 s]

[INFO] Kogito Runtimes Build BOM .......................... FAILURE [  2.751 s]

[INFO] ------------------------------------------------------------------------

[INFO] BUILD FAILURE

[INFO] ------------------------------------------------------------------------

[INFO] Total time:  4.223 s

[INFO] Finished at: 2021-02-03T12:31:59-05:00

[INFO] ------------------------------------------------------------------------

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.8:run (default) on project kogito-build-parent: An Ant BuildException has occured: Error while creating report

[ERROR] around Ant part ...<report>... @ 8:11 in /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/kogito-build-parent/target/antrun/build-main.xml: Error while analyzing kogito-quarkus-extension/integration-test-hot-reload/target/classes/org/kie/kogito/app/DecisionModels.class. Can't add different class with same name: org/kie/kogito/app/DecisionModels

[ERROR] -> [Help 1]

org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.8:run (default) on project kogito-build-parent: An Ant BuildException has occured: Error while creating report

around Ant part ...<report>... @ 8:11 in /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/kogito-build-parent/target/antrun/build-main.xml

    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
```

Jenkins retest this

**Comment by tarilabs** [comment]
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:3.0.1:integration-test (default-integration-test) on project kogito-springboot-archetype: 
[ERROR] Archetype IT 'it-basic' failed: Execution failure: exit code = 1
[ERROR] -> [Help 2]
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project process-management-springboot-addon: Compilation failure: Compilation failure: 
[ERROR] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/process-management/process-management-springboot-addon/src/main/java/org/kie/kogito/process/management/exceptions/ExceptionsHandler.java:[28,32] package org.springframework.http does not exist
[ERROR] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/process-management/process-management-springboot-addon/src/main/java/org/kie/kogito/process/management/exceptions/ExceptionsHandler.java:[29,32] package org.springframework.http does not exist
[ERROR] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/process-management/process-management-springboot-addon/src/main/java/org/kie/kogito/process/management/exceptions/ExceptionsHandler.java:[30,32] package org.springframework.http does not exist
[ERROR] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/process-management/process-management-springboot-addon/src/main/java/org/kie/kogito/process/management/exceptions/ExceptionsHandler.java:[31,47] package org.springframework.web.bind.annotation does not exist
[ERROR] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/process-management/process-management-springboot-addon/src/main/java/org/kie/kogito/process/management/exceptions/ExceptionsHandler.java:[32,47] package org.springframework.web.bind.annotation does not exist
[ERROR] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/process-management/process-management-springboot-addon/src/main/java/org/kie/kogito/process/management/exceptions/ExceptionsHandler.java:[34,2] cannot find symbol
[ERROR]   symbol: class ControllerAdvice
[ERROR] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/process-management/process-management-springboot-addon/src/main/java/org/kie/kogito/process/management/exceptions/ExceptionsHandler.java:[35,61] cannot find symbol
[ERROR]   symbol: class ResponseEntity
```

🎱 🤷‍♂️ 

Jenkins retest this

File: kogito-build-parent/pom.xml [hunk 1]
```diff
@@ -1363,6 +1363,8 @@
                               <include name="**/target/classes/**/*.class"/>
                               <!-- To avoid a complaint about duplicate classes from archetypes. -->
                               <exclude name="**/target/test-classes/**/*.class"/>
+                              <!-- To avoid a complaint about duplicate classes from archetype of Quarkus (test-compile) -->
+                              <exclude name="**/target/test-compile/**/*.class"/>
```
Comment (tarilabs) [inline kogito-build-parent/pom.xml:1367]: @rsynek

**Review by rsynek — APPROVED** [review]

File: kogito-build-parent/pom.xml (hunk 1 above)
Comment (rsynek) [reply kogito-build-parent/pom.xml:1367]: As agreed in a call, let's see if this exclusion alone helps. For the others: the `test-compile` directory of archetypes contains in runtime the included `**/target/classes` structure.
If we still see duplicated from the hot reload test, let's add one more exclude:
`<exclude name="**/integration-test-hot-reload/**/target/classes/**/*.class"/>`.

**Comment by tarilabs** [comment]
```
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ mongodb-persistence-addon ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 12 source files to /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/persistence/mongodb-persistence-addon/target/test-classes
[WARNING] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/persistence/mongodb-persistence-addon/src/test/java/org/kie/kogito/mongodb/MongoDBProcessInstancesIT.java:[44,40] org.kie.kogito.services.identity.StaticIdentityProvider in org.kie.kogito.services.identity has been deprecated
[WARNING] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/persistence/mongodb-persistence-addon/src/test/java/org/kie/kogito/mongodb/MongoDBProcessInstancesIT.java:[60,67] org.kie.kogito.services.identity.StaticIdentityProvider in org.kie.kogito.services.identity has been deprecated
[WARNING] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-runtimes/addons/persistence/mongodb-persistence-addon/src/test/java/org/kie/kogito/mongodb/DocumentProcessInstanceMarshallerTest.java:[108,85] non-varargs call of varargs method with inexact argument type for last parameter;
  cast to org.kie.api.marshalling.ObjectMarshallingStrategy for a varargs call
  cast to org.kie.api.marshalling.ObjectMarshallingStrategy[] for a non-varargs call and to suppress this warning
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ mongodb-persistence-addon ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.kie.kogito.mongodb.MongoDBProcessInstancesTest
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
```

It is stuck at here

Jenkins retest this

**Comment by tarilabs** [comment]
I guess it worked --> https://github.com/kiegroup/kogito-runtimes/pull/1035#issuecomment-773217301

now we need to hope as I see all other PRs failing in a later pipeline stage.. we will see

**Comment by tarilabs** [comment]
```
[INFO] Kogito :: JIT Executor DMN ......................... FAILURE [  0.345 s]
[INFO] Kogito :: JIT Executor Runner ...................... SKIPPED
[INFO] Kogito Apps :: Integration Tests ................... SUCCESS [  0.016 s]
[INFO] Kogito Apps :: Integration Tests :: Jobs Service ... SUCCESS [  0.030 s]
[INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Common SUCCESS [  0.290 s]
[INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Quarkus SUCCESS [ 43.636 s]
[INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Spring Boot SUCCESS [ 28.725 s]
[INFO] Kogito Apps :: Integration Tests :: Trusty Service . SUCCESS [  0.023 s]
[INFO] Kogito Apps :: Integration Tests :: Trusty Service :: Common SUCCESS [  0.337 s]
[INFO] Kogito Apps :: Integration Tests :: Trusty Service :: Quarkus SUCCESS [02:12 min]
[INFO] Kogito Apps :: Integration Tests :: Trusty Service :: Spring Boot SUCCESS [01:52 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  37:53 min
[INFO] Finished at: 2021-02-04T07:01:51-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project jitexecutor-dmn: Compilation failure: Compilation failure: 
[ERROR] /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes/kogito-apps/jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/JITDMNServiceImpl.java:[32,31] cannot find symbol
[ERROR]   symbol:   class DMNResult
[ERROR]   location: package org.kie.kogito.dmn.rest
```

The SonarCloud analysis seems solved.
The new failure is due to the fact that, allegedly, repos are not checked out at-once by Jenkins, so it passes minutes between a clone of the repo and the other, leading to this kind of problems if a PR is merged in the meantime.

Jenkins retest this

**Comment by tarilabs** [comment]
Jenkins retest this

**Comment by tarilabs** [comment]
```
2 hr 56 min ago
```

and counting.

Jenkins retest this

**Comment by tarilabs** [comment]
Run 2986 is GREEN 

https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/kogito/job/ghprb-webhooks/job/kogito-runtimes/2986

which is the latest, is green, but on GitHub it shows RED 🤦‍♂️ 🤷‍♂️ 🎱 

I believe this red is caused by ELDER runs which have timed-out (like [2981](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/kogito/job/ghprb-webhooks/job/kogito-runtimes/2981/) and so they REGARDLESS update the status even if more RECENT job runs have completed successfully.

Jenkins retest this

**Comment by tarilabs** [comment]
after chasing around for things which are not very related to the scope of this PR, this is finally green 🙆‍♂️ 
I would like to proceed swiftly to merge,
so kindly @jomarko @danielezonca if you can approve accordingly please.

File: archetypes/kogito-springboot-archetype/src/main/resources/archetype-resources/src/test/java/TrafficViolationTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
```
Comment (danielezonca) [inline archetypes/kogito-springboot-archetype/src/main/resources/archetype-resources/src/test/java/TrafficViolationTest.java:17]: Unused?

**Review by danielezonca — APPROVED** [review]

**Pull request merged**